packages/services/src/kernel/default.ts

```diff
@@ -1527,11 +1532,13 @@ namespace Private {
     // We might want to move the handleRestart to after we get the response back
 
     // Handle the restart on all of the kernels with the same id.
-    each(runningKernels, k => {
-      if (k.id === kernel.id) {
-        k.handleRestart();
-      }
-    });
+    await Promise.all([
```

Comment:
It seems like the promises are in a nested array. I think there is an extra `[` here.

Reply:
Good catch!